deriva/core/ermrest_model.py

```diff
@@ -709,11 +742,7 @@ def define_asset(cls,
           """
 
         if hatrac_template is None:
-            hatrac_template = '/hatrac/%s/%s/{{{MD5}}}.{{#encode}}{{{Filename}}}{{/encode}}' % (sname, tname)
-
-        def add_asset_annotations(custom):
-            annotations.update(custom)
-            return annotations
```

Comment:
I think this function should not have been removed, but altered to be something like:
```
def add_asset_annotations(custom):
    annotations = tag.table_display: {
        'row_name': {
            'row_markdown_pattern': '{{{Filename}}}'
        }
    }
    annotations.update(custom)
    return annotations
```
to fit with the pattern used for the other fields and keyword args.

As an aside, we could probably start using the `|` or `|=` operators to merge dicts, unless we still want to support older Python versions. However, our codebases are riddled with the older idiom of calling dict.update() explicitly since that was the only way for much of its history...

Reply:
done